view/SSViewer.php

```diff
@@ -1106,9 +1111,11 @@ public function process($item, $arguments = null, $inheritedScope = null) {
 		// through $Content and $Layout placeholders.
 		foreach(array('Content', 'Layout') as $subtemplate) {
 			if(isset($this->chosenTemplates[$subtemplate])) {
-				$subtemplateViewer = new SSViewer($this->chosenTemplates[$subtemplate], $this->parser);
+				$subtemplateViewer = clone $this;
+				// Disable requirements - this will be handled by the parent template
 				$subtemplateViewer->includeRequirements(false);
-				$subtemplateViewer->setPartialCacheStore($this->getPartialCacheStore());
+				// The subtemplate is the only file we want to process, so set it as the "main" template file
+				$subtemplateViewer->chosenTemplates = array('main' => $this->chosenTemplates[$subtemplate]);
```

Comment:
An alternative here would be to add a method to clear the chosen templates (`$this->resetTemplates()` or something) and then use `setTemplateFile('main', $this->chosenTemplates[$subtemplate])`. I can do that if that would be preferred.